app/src/main/java/com/example/series_collector/ui/detail/DetailViewModel.kt

```diff
@@ -0,0 +1,86 @@
+package com.example.series_collector.ui.detail
+
+
+import androidx.lifecycle.*
+import androidx.paging.PagingData
+import androidx.paging.cachedIn
+import com.example.series_collector.data.Series
+import com.example.series_collector.data.api.PageInfo
+import com.example.series_collector.data.api.SeriesVideo
+import com.example.series_collector.data.repository.SeriesFollowedRepository
+import com.example.series_collector.data.repository.SeriesRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+
+@HiltViewModel
+class DetailViewModel @Inject constructor(
+    savedStateHandle: SavedStateHandle,
+    private val seriesFollowedRepository: SeriesFollowedRepository,
+    private val seriesRepository: SeriesRepository,
+) : ViewModel() {
+
+    val seriesId: String = savedStateHandle.get<String>(SERIES_ID_SAVED_STATE_KEY)!!
+
+    val isFollowed = seriesFollowedRepository.isFollowed(seriesId).asLiveData()
+
+    private val _isLoading = MutableLiveData<Boolean>()
+    val isLoading: LiveData<Boolean> = _isLoading
+
+    private val _series = MutableLiveData<Series>()
+    val series: LiveData<Series> = _series
+
+    private val _seriesPageInfo = MutableLiveData<PageInfo>()
+    val seriesPageInfo: LiveData<PageInfo> = _seriesPageInfo
+
+    private var currentQueryValue: String? = null
+    private var currentSearchResult: Flow<PagingData<SeriesVideo>>? = null
+
+
+    init {
+        _isLoading.value = true
+        viewModelScope.launch {
+            seriesRepository.getSeriesStream(seriesId)
+                .map {
+                    _series.value = it
+                    _isLoading.value = false
+                }
+                .launchIn(viewModelScope)
+
+            _seriesPageInfo.value =
+                seriesRepository.getPageInfo(seriesId)
+
+        }
```

Comment:
```suggestion
        seriesRepository.getSeriesStream(seriesId)
            .onEach {
                _series.value = it
                _isLoading.value = false
            }
            .launchIn(viewModelScope)

        viewModelScope.launch {
            _seriesPageInfo.value =
                seriesRepository.getPageInfo(seriesId)
        }
```
flow transform은 launchIn 함수를 통해 동작할 scope를 지정하므로 CoroutineScope내부에서 호출 할 필요가 없습니다 😄

Reply:
넵  launchIn 함수 좀 다시보겠습니다!